### PR TITLE
SPARK-SNOWFLAKE-251

### PR DESCRIPTION
**Pull request by YevIgn**
Suggested fix approach.

### CODE REVIEW

**Comment by YevIgn** [comment]
For the ticket - https://github.com/snowflakedb/spark-snowflake/issues/251

**Comment by YevIgn** [comment]
Not sure how to run integration tests properly, created PR for reference purposes mostly.

**Comment by sfc-gh-mrui** [comment]
Hi, @YevIgn, Thanks let us know. We will take a look on it.

File: src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/BooleanStatement.scala [hunk 1]
```diff
@@ -31,8 +31,13 @@ private[querygeneration] object BooleanStatement {
 
     Option(expr match {
       case In(child, list) if list.forall(_.isInstanceOf[Literal]) =>
-        convertStatement(child, fields) + "IN" +
-          blockStatement(convertStatements(fields, list: _*))
+        list match {
```
Comment (sfc-gh-mrui) [inline src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/BooleanStatement.scala:34]: Can you add unit test for this case too?

**Comment by sfc-gh-mrui** [comment]
I have added explanation to issue description: https://github.com/snowflakedb/spark-snowflake/issues/251

**Pull request closed without merging**